sentry/src/main/java/io/sentry/Sentry.java

```diff
@@ -254,47 +258,56 @@ private static synchronized void init(
-            if (!options.isEnableStartupProfiling()) {
-              return;
-            }
-            if (!options.isTracingEnabled()) {
-              options
-                  .getLogger()
-                  .log(
-                      SentryLevel.INFO,
-                      "Tracing is disabled and startup profiling will not start.");
-              return;
-            }
-            try {
-              if (startupProfilingConfigFile.createNewFile()) {
-                final @NotNull TracesSamplingDecision startupSamplingDecision =
-                    sampleStartupProfiling(options);
-                final @NotNull SentryStartupProfilingOptions startupProfilingOptions =
-                    new SentryStartupProfilingOptions(options, startupSamplingDecision);
-                try (Writer fileWriter = new SentryFileWriter(startupProfilingConfigFile)) {
-                  options.getSerializer().serialize(startupProfilingOptions, fileWriter);
+    try {
+      sentryExecutorService.submit(
+          () -> {
+            final String cacheDirPath = options.getCacheDirPathWithoutDsn();
+            if (cacheDirPath != null) {
+              final @NotNull File startupProfilingConfigFile =
+                  new File(cacheDirPath, STARTUP_PROFILING_CONFIG_FILE_NAME);
+              // We always delete the config file for startup profiling
+              FileUtils.deleteRecursively(startupProfilingConfigFile);
+              if (!options.isEnableStartupProfiling()) {
+                return;
+              }
+              if (!options.isTracingEnabled()) {
+                options
+                    .getLogger()
+                    .log(
+                        SentryLevel.INFO,
+                        "Tracing is disabled and startup profiling will not start.");
+                return;
+              }
+              try {
+                if (startupProfilingConfigFile.createNewFile()) {
+                  final @NotNull TracesSamplingDecision startupSamplingDecision =
+                      sampleStartupProfiling(options);
+                  final @NotNull SentryStartupProfilingOptions startupProfilingOptions =
+                      new SentryStartupProfilingOptions(options, startupSamplingDecision);
+                  try (Writer fileWriter = new SentryFileWriter(startupProfilingConfigFile)) {
+                    options.getSerializer().serialize(startupProfilingOptions, fileWriter);
+                  }
                 }
+              } catch (IOException e) {
+                options
+                    .getLogger()
+                    .log(SentryLevel.ERROR, "Unable to create startup profiling config file. ", e);
               }
-            } catch (IOException e) {
-              options
-                  .getLogger()
-                  .log(SentryLevel.ERROR, "Unable to create startup profiling config file. ", e);
             }
-          }
-        });
+          });
+    } catch (RejectedExecutionException e) {
```

Comment:
```suggestion
    } catch (Throwable e) {
```
let's make it bulletproof to everything?

Reply:
@stefanosiano this is still a `RejectedExeuctionException` ;)